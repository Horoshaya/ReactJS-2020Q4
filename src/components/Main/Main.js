import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';

import MovieList from '../MovieList/MovieList';
import { FiltersSortBar } from '../FiltersSortBar/FiltersSortBar';
import { getMovie, sortByGenre } from '../../store/actions/actionCreators';

export const MainContext = React.createContext();

const Main = () => {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getMovie());
  }, [dispatch]);

  const sortMovieByGenre = (genre) => {
    dispatch(sortByGenre(genre, movieReducer));
  };

  const movieReducer = useSelector((store) => store.movieReducer);

  return (
    <main>
      <MainContext.Provider value={sortMovieByGenre}>
        <FiltersSortBar />
      </MainContext.Provider>
      <MovieList movieReducer={movieReducer} />
    </main>
  );
};

export default Main;
