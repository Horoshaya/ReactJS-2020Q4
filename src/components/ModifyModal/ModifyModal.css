.title {
  text-transform: uppercase;
  font-weight: 100;
}

.buttonsWrapper {
  display: flex;
  justify-content: space-between;
}

.reset,
.save {
  padding: 10px 15px;
  text-transform: uppercase;
  border: 1px solid #f65261;
  border-radius: 4px;
  cursor: pointer;
}

.reset {
  color: #f65261;
  background: #232323;
}

.save {
  color: #ffffff;
  background: #f65261;
}
