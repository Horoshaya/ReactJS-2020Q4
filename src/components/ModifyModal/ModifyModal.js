import React from 'react';
import PropTypes from 'prop-types';

import Modal from '../Modal/Modal';
import TextField from '../TextField/TextField';
import styles from './ModifyModal.css';

const ModifyModal = ({ triggerModal, movieInfo }) => {
  return (
    <Modal triggerModal={() => triggerModal('modify')}>
      <form>
        <h2 className={styles.title}>edit movie</h2>
        <TextField label="movie id" title={movieInfo.id} />
        <TextField label="title" title={movieInfo.title} />
        <TextField label="url" title={movieInfo.imgUrl} />
        <div className={styles.buttonsWrapper}>
          <button className={styles.reset}>reset</button>
          <button className={styles.save}>save</button>
        </div>
      </form>
    </Modal>
  );
};

ModifyModal.propTypes = {
  type: PropTypes.string,
  movieInfo: PropTypes.shape({
    id: PropTypes.number,
    title: PropTypes.string,
    releaseDate: PropTypes.string,
    url: PropTypes.string,
    overview: PropTypes.string,
    overview: PropTypes.string,
  }),
};

ModifyModal.defaultProps = {
  type: '',
  movieInfo: {
    id: null,
    title: '',
    releaseDate: '',
    url: '',
    genre: '',
    overview: '',
    runtime: '',
  },
};

export default ModifyModal;
