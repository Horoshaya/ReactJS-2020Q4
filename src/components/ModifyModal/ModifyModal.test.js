import React from 'react';
import rerender from 'react-test-renderer';
import ModifyModal from './ModifyModal';
import { MainContext } from '../Main/Main';

const mockDispatch = jest.fn();
jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: () => mockDispatch,
}));

describe('when ModifyModal', () => {
  test('then snapshot created', () => {
    const element = rerender.create(
      <MainContext.Provider value={'test'}>
        <ModifyModal />
      </MainContext.Provider>,
    );
    const tree = element.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
