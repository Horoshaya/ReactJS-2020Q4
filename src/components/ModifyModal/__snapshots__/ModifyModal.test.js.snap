// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when ModifyModal then snapshot created 1`] = `
<div>
  <div>
    <span
      onClick={[Function]}
    />
    <form
      action="#"
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <h2>
        edit movie
      </h2>
      <label>
        movie id
        <input
          name="id"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
        />
      </label>
      <label>
        title
        <input
          name="title"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value=""
        />
      </label>
      <label>
        release date
        <input
          name="release_date"
          onBlur={[Function]}
          onChange={[Function]}
          type="date"
        />
      </label>
      <label>
        url
        <input
          name="poster_path"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
        />
      </label>
      <label>
        genre
        <select
          multiple={true}
          name="genres"
          onChange={[Function]}
          size="10"
          value={Array []}
        >
          <option
            label="Drama"
            value="Drama"
          />
          <option
            label="Documentary"
            value="Documentary"
          />
          <option
            label="Comedy"
            value="Comedy"
          />
          <option
            label="Horror"
            value="Horror"
          />
          <option
            label="Crime"
            value="Crime"
          />
          <option
            label="Romance"
            value="Romance"
          />
          <option
            label="Adventure"
            value="Adventure"
          />
          <option
            label="Action"
            value="Action"
          />
          <option
            label="Fantasy"
            value="Fantasy"
          />
          <option
            label="Science Fiction"
            value="Science Fiction"
          />
        </select>
      </label>
      <label>
        overview
        <input
          name="overview"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value=""
        />
      </label>
      <label>
        vote_average
        <input
          name="vote_average"
          onBlur={[Function]}
          onChange={[Function]}
          type="number"
        />
      </label>
      <label>
        runtime
        <input
          name="runtime"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value=""
        />
      </label>
      <div>
        <button>
          reset
        </button>
        <button
          type="submit"
        >
          save
        </button>
      </div>
    </form>
  </div>
  <div />
</div>
`;
